### PR TITLE
snap: rename gdbserver option to `snap run --gdbserver`

### PR DESCRIPTION
**Pull request by mvo5**
The `snap run --experimental-gdbserver` option is available
since a while and we got no negative feedback so far. Given
that it's also superior to the `snap run --gdb` option this
PR make it available with the *experimental* prefix.

The old `snap run --experimental-gdbserver` option is still
available for compatbility and can be removed after some
releases.

### CODE REVIEW

**Comment by pedronis** [comment]
@mvo did we understand if this looks for gdbserver in the right places after the confusion in #snappy?

**Comment by mvo5** [comment]
@pedronis Excellent point about gdbserver - I pushed https://github.com/snapcore/snapd/pull/9912 that provides a more useful error message when gdbserver is not installed on the host.

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -75,10 +75,11 @@ type cmdRun struct {
...
 	// can also carry extra options for strace. This is why there is
 	// "default" and "optional-value" to distinguish this.
-	Strace    string `long:"strace" optional:"true" optional-value:"with-strace" default:"no-strace" default-mask:"-"`
-	Gdb       bool   `long:"gdb"`
-	Gdbserver string `long:"experimental-gdbserver" default:"no-gdbserver" optional-value:":0" optional:"true"`
-	TraceExec bool   `long:"trace-exec"`
+	Strace                string `long:"strace" optional:"true" optional-value:"with-strace" default:"no-strace" default-mask:"-"`
+	Gdb                   bool   `long:"gdb"`
+	Gdbserver             string `long:"gdbserver" default:"no-gdbserver" optional-value:":0" optional:"true"`
+	ExperimentalGdbserver string `long:"experimental-gdbserver" default:"no-gdbserver" optional-value:":0" optional:"true" hidden:"yes"`
```
Comment (bboozzoo) [inline cmd/snap/cmd_run.go:81]: go-flags seems to have some aliasing support, though the docs states it works on commands https://pkg.go.dev/github.com/jessevdk/go-flags#hdr-Available_field_tags but maybe worth trying

File: tests/main/snap-run-gdbserver/task.yaml [hunk 2]
```diff
@@ -11,12 +11,12 @@ restore: |
...
 
 execute: |
-    echo "Test snap run --experimental-gdbserver works"
+    echo "Test snap run --gdbserver works"
 
     # run the gdbserver command as a user
     # XXX: use "systemd-run --user -p StandardOutput=file:$(pwd)/stdout"
     # shellcheck disable=SC2016
-    su -c 'snap run --experimental-gdbserver test-snapd-sh.sh -c "echo hello-hello-hello uid:$UID"' test >stdout &
+    su -c 'snap run --gdbserver test-snapd-sh.sh -c "echo hello-hello-hello uid:$UID"' test >stdout &
```
Comment (bboozzoo) [inline tests/main/snap-run-gdbserver/task.yaml:19]: We should probably do this basic test with --experimental-gdbserver and --gdbserver (maybe via environment flag) for as long as we'll have the flag in `snap run`.

**Review by bboozzoo — COMMENTED** [review]
LGTM

**Review by bboozzoo — APPROVED** [review]
LGTM. Thanks!

**Review by anonymouse64 — APPROVED** [review]
lgtm, but the note to translators is confusing/erroneous :smile:

**Pull request merged**